app/src/utils/walletStore.ts

```diff
@@ -1,393 +1,350 @@
-
-	if (walletName) {
-		walletNameStore.updateName(walletName);
-	}
+async function connect(): Promise<void> {
+    const { connected, connecting, disconnecting, wallet, ready, adapter } = get(walletStore);
+    if (connected || connecting || disconnecting) return;
+
+    if (!adapter) throw newError(new WalletNotSelectedError());
+
+    if (!ready) {
+        walletStore.resetWallet();
+
+        if (typeof window !== 'undefined') {
+            window.open(adapter.url, '_blank');
+        }
+
+        throw newError(new WalletNotReadyError());
+    }
+
+    try {
+        walletStore.setConnecting(true);
+        await adapter.connect();
+    } catch (error: unknown) {
+        walletStore.resetWallet();
+        throw error;
+    } finally {
+        walletStore.setConnecting(false);
+    }
 }
 
-async function select(newName: WalletName | null): Promise<void> {
-	const { walletName } = get(walletNameStore);
-	if (walletName === newName) return;
-
-	const { adapter } = get(walletAdapterStore);
-	if (adapter) await disconnect();
+function createWalletStore() {
+    const { subscribe, update } = writable<WalletStore>({
+        autoConnect: false,
+        wallets: [],
+        adapter: null,
+        connected: false,
+        connecting: false,
+        disconnecting: false,
+        localStorageKey: 'walletAdapter',
+        onError: (error: WalletError) => console.error(error),
+        publicKey: null,
+        ready: "NotDetected" as WalletReadyState, //?? is this correct?
+        wallet: null,
+        name: null,
+        walletsByName: {},
+        connect,
+        disconnect,
+        select,
+        sendTransaction,
+        signTransaction: undefined,
+        signAllTransactions: undefined,
+        signMessage: undefined,
+    });
+
+    function updateWalletState(adapter: Adapter | null) {
+        updateAdapter(adapter);
+        update((store) => ({
+            ...store,
+            name: adapter?.name || null,
+            wallet: adapter,
+            // ready: false,
+			ready: adapter?.readyState,
+            publicKey: adapter?.publicKey || null,
+            connected: adapter?.connected || false,
+        }));
+
+        // if (!(wallet?.name && adapter)) return;
+        if (!adapter) return;
+
+		if (shouldAutoConnect()) {
+			autoConnect();
+		}
 
-	walletNameStore.updateName(newName);
+        // Asynchronously update the ready state
+        // const waiting = wallet.name;
+        // (async function () {
+        //     const ready = await adapter.ready();
+        //     // If the selected wallet hasn't changed while waiting, update the ready state
+        //     if (wallet.name === waiting) {
+        //         update((store) => ({
+        //             ...store,
+        //             ready,
+        //         }));
+
+        //         if (shouldAutoConnect()) {
+        //             autoConnect();
+        //         }
+        //     }
+        // })();
+    }
```

Comment:
Not needed anymore since we have the ready property linked to the adapter one, thanks to the `onReadyStateChange` function.